sylvia-derive/src/message.rs

```diff
@@ -729,16 +756,15 @@ impl<'a> GlueMessage<'a> {
                     where D: serde::Deserializer<'de>,
                 {
                     use serde::de::Error;
+
                     let val = #sylvia ::serde_value::Value::deserialize(deserializer)?;
 
                     #(#deserialization_attempts)*
                     #contract_deserialization_attempt
 
-                    Err(D::Error::custom(format!(
-                        "Expected any of {} or {} messages, but cannot deserialize to neither of those\n{}",
+                    Err(D::Error::custom(format!("Expected any of {} or {} messages, but neither of these was provided.",
```

Comment:
As I understand properly, you would print here like something like:

> Expected any of Cw1 or ExecMsg messages, but neither of those was provided.

Not very helpful (especially `ExecMsg` part). Better would be:

> Unsupported message received. Messages supported by this contract: `modify_admins`, `freeze`, `execute`.

We have all messages here, so just collect them with [`concat`](https://doc.rust-lang.org/std/primitive.slice.html#method.concat) to a single vector, and then cal [`join`](https://doc.rust-lang.org/std/primitive.slice.html#method.join) on it. It is an additional allocation, but it is an error case - a way less relevant one (better error messages > avoiding every allocation possible). And even this allocation can be avoided (it is preferred here as it is generated so a bit of noise does not do harm): make an array `messages()` results, and do something like:
```rust

let mut msg = all_messages
  .into_iter()
  .flat_map(|messages| messages.iter())
  .fold("Unsupported message received. Messages supported by this contract: ".to_owned(), |mut acc, message|  {
    acc.push_str(message); 
    acc.push_str(", ");
  });
  msg.truncate(msg.len() - 2);
```

Possibly with some special care if there are no messages at all.